numba/np/ufunc/gufunc.py

```diff
@@ -26,6 +28,30 @@ def __init__(self, py_func, signature, identity=None, cache=None,
             py_func, signature, identity, cache, targetoptions)
         self.__name__ = self.gufunc_builder.py_func.__name__
 
+    def _reduce_states(self):
+        gb = self.gufunc_builder
+        dct = dict(
+            py_func=gb.py_func,
+            signature=gb.signature,
+            identity=self._identity,
+            cache=gb.cache,
+            is_dynamic=self._is_dynamic,
+            targetoptions=gb.targetoptions,
+            typesigs=gb._sigs,
```

Comment:
Think `self._frozen` should be in here, whilst it has no impact on dynamic gufunc, it is transmittable state.

Reply:
fixed